### PR TITLE
Make lastly used side panel item persisted across re-launch

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/34399

Resolves 

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

`SidebarBrowserTest.LastlyUsedSidePanelItemTest`
Manual test - refer to linked issue

### CODE REVIEW

File: browser/ui/sidebar/sidebar_utils.cc [hunk 1]
```diff
@@ -158,4 +164,46 @@ absl::optional<SidebarItem> AddItemForSidePanelIdIfNeeded(Browser* browser,
...
+  }
+
+  prefs->SetInteger(kLastlyUsedBuiltInItemType, static_cast<int>(type));
+}
+
+absl::optional<SidePanelEntryId> GetLastlyUsedSidePanel(Browser* browser) {
+  PrefService* prefs = browser->profile()->GetPrefs();
+  BuiltInItemType type = static_cast<BuiltInItemType>(
+      prefs->GetInteger(kLastlyUsedBuiltInItemType));
+  if (type == BuiltInItemType::kNone) {
```
Comment (fallaciousreasoning) [inline browser/ui/sidebar/sidebar_utils.cc:196]: Optional: Is this case needed?

File: components/sidebar/pref_names.h [hunk 2]
```diff
@@ -16,6 +16,8 @@ inline constexpr char kSidebarShowOption[] =
 inline constexpr char kSidebarItemAddedFeedbackBubbleShowCount[] =
     "brave.sidebar.item_added_feedback_bubble_shown_count";
 inline constexpr char kSidePanelWidth[] = "brave.sidebar.side_panel_width";
+inline constexpr char kLastlyUsedBuiltInItemType[] =
```
Comment (fallaciousreasoning) [inline components/sidebar/pref_names.h:19]: nit:
```suggestion
inline constexpr char kLastUsedBuiltInItemType[] =
```

File: components/sidebar/pref_names.h [hunk 3]
```diff
@@ -16,6 +16,8 @@ inline constexpr char kSidebarShowOption[] =
...
+    "brave.sidebar.lastly_used_built_in_item_type";
```
Comment (fallaciousreasoning) [inline components/sidebar/pref_names.h:20]:
```suggestion
    "brave.sidebar.last_used_built_in_item_type";
```

**Review by fallaciousreasoning — APPROVED** [review]
LGTM

File: components/sidebar/pref_names.h (hunk 2 above)
Comment (simonhong) [reply components/sidebar/pref_names.h:19]: All renamed 👍🏼

File: browser/ui/sidebar/sidebar_utils.cc (hunk 1 above)
Comment (simonhong) [reply browser/ui/sidebar/sidebar_utils.cc:196]: Yes, need this as `BuiltInItemType::kNone` is initial value for `kLastUsedBuiltInItemType` prefs.

Comment (fallaciousreasoning) [reply browser/ui/sidebar/sidebar_utils.cc:196]: Won't it be handled by this check?

```cpp
if (!static_cast<BraveBrowser*>(browser)
           ->sidebar_controller()
           ->model()
           ->GetIndexOf(type)) {
    return absl::nullopt;
  }
```

Comment (simonhong) [reply browser/ui/sidebar/sidebar_utils.cc]: ahaaa you're right :) Done!

Comment (simonhong) [reply browser/ui/sidebar/sidebar_utils.cc]: Oops, I found a existing bug and added fix(https://github.com/brave/brave-core/pull/21022/commits/4cb5007e8a9ad22f522ba0034540b8aa94bcf18b)

**Review by fallaciousreasoning — APPROVED** [review]
Still LGTM

**Pull request merged**